nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestFTPCharset.java

```diff
@@ -0,0 +1,295 @@
+                arguments("folder3", "اختبار"),
+                arguments("اختبار", "folder4"),
+                arguments("folder4", "Госагїzатїой"),
+                arguments("Госагїzатїой", "folder5"),
+                arguments("folder5", "し回亡丹し工z丹卞工回几"),
+                arguments("し回亡丹し工z丹卞工回几", "folder6")
+        );
+    }
+
+    public static Stream<String> filenamesProvider() {
+        return Stream.of(
+                "1.txt",
+                "æøå.txt",
+                "اختبار.txt",
+                "Госагїzатїой.txt",
+                "し回亡丹し工z丹卞工回几.txt");
+    }
+
+    /**
+     * Start test FTP server (if using embedded).
+     *
+     * @throws FtpException on server startup failure
+     */
+    @BeforeAll
+    static void beforeAll() throws FtpException {
+        if (EMBED_FTP_SERVER) {
+            FOLDER.mkdirs();
+            // setup ftp user
+            final PropertiesUserManagerFactory userManagerFactory = new PropertiesUserManagerFactory();
+            userManagerFactory.setUrl(TestFTPCharset.class.getClassLoader()
+                    .getResource("TestFTPCharset/users.properties"));
+            userManagerFactory.setPasswordEncryptor(new ClearTextPasswordEncryptor());
+            final UserManager userManager = userManagerFactory.createUserManager();
+            final BaseUser ftpuser = (BaseUser) userManager.getUserByName(USER);
+            ftpuser.setHomeDirectory(FOLDER.getAbsolutePath());
+            userManager.save(ftpuser);
+            // setup embedded ftp server
+            final FtpServerFactory serverFactory = new FtpServerFactory();
+            serverFactory.setUserManager(userManager);
+            final FileSystemFactory fileSystemFactory = serverFactory.getFileSystem();
+            final FileSystemView view = fileSystemFactory.createFileSystemView(ftpuser);
+            final FtpFile workingDirectory = view.getWorkingDirectory();
+            final Object physicalFile = workingDirectory.getPhysicalFile();
+            assertTrue(physicalFile instanceof File);
+            assertEquals(FOLDER.getAbsolutePath(), ((File) physicalFile).getAbsolutePath());
+            final ListenerFactory factory = new ListenerFactory();
+            factory.setPort(PORT);
+            serverFactory.addListener("default", factory.createListener());
+            FTP_SERVER = serverFactory.createServer();
+            FTP_SERVER.start();
+        }
+    }
+
+    /**
+     * Stop test FTP server (if using embedded).
+     *
+     * @throws IOException on failure to clean up test data
+     */
+    @AfterAll
+    static void afterAll() throws IOException {
+        if (EMBED_FTP_SERVER) {
+            FTP_SERVER.stop();
+            while (!FTP_SERVER.isStopped()) {
+                try {
+                    Thread.sleep(100L);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            FileUtils.deleteDirectory(FOLDER);
```

Comment:
Refactoring to use JUnit 5 TempDir should remove the need for deleting the directory manually, although it might be necessary to clear the directory, depending on the approach.